File: Algorithm/Programmers/Python/Level 2/짝지어 제거하기.py
def solution(S):
    stk=[]
    for s in S:
        if stk:
            if stk[-1]==s:
                stk.pop()
            else:
                stk.append(s)
        else:
            stk.append(s)
    if stk:
        return 0
    else:
        return 1
        

File: Algorithm/Programmers/Java/Level 2/짝지어_제거하기.java
import java.util.Stack;
class Solution
{
    Stack<Character> stk = new Stack();
    public int solution(String s)
    {
        for(char c : s.toCharArray()) {
            if(stk.empty()) {
                stk.add(c);
                continue;
            }
            
            if(stk.peek().equals(c)) {
                stk.pop();
                continue;
            }
                       
            stk.add(c);
        }

        return stk.empty() ? 1 : 0;
    }
}